satellite/marketing/service.go

```diff
@@ -78,14 +78,27 @@ func (s *Service) InsertNewOffer(ctx context.Context, offer *NewOffer) (o *Offer
 	return o, nil
 }
 
-// UpdateOffer modifies an existing offer in the db when the offer status is set to NoStatus
-func (s *Service) UpdateOffer(ctx context.Context, offer *UpdateOffer) (err error) {
+// RedeemOffer adds 1 to the number of redeemed for an offer
+func (s *Service) RedeemOffer(ctx context.Context, uo *UpdateOffer) (err error) {
 	defer mon.Task()(&ctx)(&err)
 
-	if offer.Status == Default {
-		offer.NumRedeemed = 0
+	if uo.Status == Default {
+		return nil
+	}
+
+	err = s.db.Offers().Redeem(ctx, uo.ID)
+	if err != nil {
+		return Error.Wrap(err)
 	}
-	err = s.db.Offers().Update(ctx, offer)
+
+	return nil
+}
+
+// FinishOffer updates an active offer's status to be Done and its expiration time to be now
+func (s *Service) FinishOffer(ctx context.Context, oID int) (err error) {
```

Comment:
https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters

Reply:
Thanks for sharing the link. I used the named result bc I needed to pass the `error` into the monkit and that has been the patter for doing so. I can definitely change it